### PR TITLE
Garrett's test driven model Danebook

### PR DESCRIPTION
**Pull request by gweinert**

### CODE REVIEW

**Comment by gweinert** [comment]
Garrett's model tested danebook

File: spec/factories/user_factory.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FactoryGirl.define do
+
+  factory :user do
+    first_name "Foo"
+    last_name "Bar"
+    email "foo@bar.com"
```
Comment (alokpradhan) [inline spec/factories/user_factory.rb]: Consider creating a unique email for every user to meet uniqueness constraint in model

File: spec/factories/like_factory.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+FactoryGirl.define do
+
+  factory :like do
+
```
Comment (alokpradhan) [inline spec/factories/like_factory.rb:4]: A user shouldn't have a like without liking something. Consider:
1. Adding presence validation to like model for liking_type and liking_id
2. Making this factory method polymorphic and setting liking type and id during creation of like

File: spec/factories/comment_factory.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+FactoryGirl.define do
+
+  factory :comment do
+    body "foo commented"
+
```
Comment (alokpradhan) [inline spec/factories/comment_factory.rb:5]: Same feedback as like

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    user.password = nil
+    expect(user).to be_invalid
+  end
+
+  it 'is not valid if password is eight spaces' do
+    user.password = "        "
+    expect(user).to be_invalid
+  end
+
+  # This should not pass either
```
Comment (alokpradhan) [inline spec/models/user_spec.rb:32]: change the values of both password and password_confirmation (as user has not yet been saved) to create a new password after save.

**Pull request closed without merging**